pkg/provisioner/ironic/ironic.go

```diff
@@ -1155,6 +1155,82 @@ func (p *ironicProvisioner) ironicHasSameImage(ironicNode *nodes.Node) (sameImag
 	return sameImage
 }
 
+func (p *ironicProvisioner) startManualCleaning(ironicNode *nodes.Node) (success bool, result provisioner.Result, err error) {
+	var cleanSteps []nodes.CleanStep
+
+	// TODO: Add manual cleaning steps for host configuration
+
+	// Start manual clean
+	if len(cleanSteps) != 0 {
+		p.log.Info("remove existing configuration and set new configuration", "steps", cleanSteps)
+		opts := nodes.ProvisionStateOpts{Target: nodes.TargetClean, CleanSteps: cleanSteps}
+		success, result, err = p.tryChangeNodeProvisionState(ironicNode, opts)
+	}
+	return
+}
+
+// Prepare remove existing configuration and set new configuration.
+// If `started` is true,  it means that we successfully executed `tryChangeNodeProvisionState`.
+func (p *ironicProvisioner) Prepare(unprepared bool) (result provisioner.Result, started bool, err error) {
+	var ironicNode *nodes.Node
+
+	if ironicNode, err = p.findExistingHost(); err != nil {
+		result, err = transientError(errors.Wrap(err, "could not find host to clean"))
+		return
+	}
+	if ironicNode == nil {
+		result, err = transientError(provisioner.NeedsRegistration)
+		return
+	}
+
+	switch nodes.ProvisionState(ironicNode.ProvisionState) {
+	case nodes.Available:
+		if unprepared {
+			result, err = p.changeNodeProvisionState(
+				ironicNode,
+				nodes.ProvisionStateOpts{Target: nodes.TargetManage},
+			)
+		}
+
+	case nodes.Manageable:
+		if unprepared {
+			started, result, err = p.startManualCleaning(ironicNode)
```

Comment:
nit: if you changed the order of the return values, you could just `return p.startManualCleaning(ironicNode)`.